R/pkg/R/serialize.R

```diff
@@ -37,24 +37,38 @@ writeObject <- function(con, object, writeType = TRUE) {
   # passing in vectors as arrays and instead require arrays to be passed
   # as lists.
```

Comment:
Yeah the `type %in% check` is fine. Lets add a comment on top of that saying `is.na only handles handles atomic vectors, lists and pairlists` (this is from the documentation)